### PR TITLE
feat(payments): add email address to download link

### LINKED ISSUE #10489: [subplat] Append User Email Address as Query Param to Product Download URL on Successful Subscription
It might be helpful to append the user's email address to the product download URL as a query param on the successful subscription confirmation page, especially for RPs with prompt=none enabled.  The post-finish-setup redirect for the passwordless subscription is already doing this.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4005)

### PR DESCRIPTION
**Pull request by IvoJP**
Because:

* MDN requested to have the email address appeneded to the download link as a query param

This commit:

* URI encodes the email address of the user and appends to the download link on a successful subscription

Closes #10489

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [X] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 1]
```diff
@@ -146,7 +146,7 @@ export const PaymentConfirmation = ({
             <a
               data-testid="download-link"
               className="button download-link"
-              href={productUrl}
+              href={productUrl + `/?email=${encodeURIComponent(email)}`}
```
Comment (dannycoates) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:149]: nit: I'd suggest not mixing concatenation and template strings /driveby

Comment (IvoJP) [reply packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx]: moved to its own variable outside of template

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 2]
```diff
@@ -60,6 +60,7 @@ export const PaymentConfirmation = ({
   );
 
   const { l10n } = useLocalization();
+  const downloadLink = productUrl + `/?email=${encodeURIComponent(email)}`;
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:63]: The productUrl is a url from the metadata, which could already have query params.  We need to accommodate that.

**Review by chenba — CHANGES_REQUESTED** [review]
We need to make it work with any url.  See comment.

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx (hunk 2 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx]: updated to handle either case - query params already present or not.

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 3]
```diff
@@ -60,6 +60,10 @@ export const PaymentConfirmation = ({
   );
 
   const { l10n } = useLocalization();
+  const downloadLink: string =
+    productUrl.indexOf('?') > -1
+      ? productUrl + `&email=${encodeURIComponent(email)}`
+      : productUrl + `?email=${encodeURIComponent(email)}`;
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:66]: I think this probably would work for most cases.  But a safer way to do this is to use the URL API: https://developer.mozilla.org/en-US/docs/Web/API/URL.  Once the url's parsed you can add an entry into its search params (https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams).

Comment (IvoJP) [reply packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx]: updated to use URL API

**Review by chenba — APPROVED** [review]
👍
